wardrobe/tradable_kitties/src/lib.rs

```diff
@@ -0,0 +1,446 @@
+//!
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use kitties::{KittyDNA, KittyData};
+use money::ConstraintCheckerError as MoneyError;
+use money::{Coin, MoneyConstraintChecker};
+use parity_scale_codec::{Decode, Encode};
+use scale_info::TypeInfo;
+use serde::{Deserialize, Serialize};
+use sp_runtime::transaction_validity::TransactionPriority;
+use sp_std::collections::btree_set::BTreeSet; // For checking the uniqueness of input and output based on dna.
+use sp_std::prelude::*;
+use tuxedo_core::{
+    dynamic_typing::{DynamicallyTypedData, UtxoData},
+    ensure, SimpleConstraintChecker,
+};
+
+#[cfg(test)]
+mod tests;
+
+/// A tradableKittyData, required for trading the kitty..
+/// It contains optional price of tradable-kitty in addition to the basic kitty data.
+#[derive(
+    Serialize,
+    Deserialize,
+    PartialEq,
+    Eq,
+    PartialOrd,
+    Ord,
+    Clone,
+    Encode,
+    Decode,
+    Hash,
+    Debug,
+    TypeInfo,
+)]
+pub struct TradableKittyData {
+    /// Basic kitty data composed from kitties piece
+    pub kitty_basic_data: KittyData,
+    /// Price of the tradable kitty
+    pub price: Option<u128>,
+}
+
+impl Default for TradableKittyData {
+    fn default() -> Self {
+        Self {
+            kitty_basic_data: KittyData::default(),
+            price: None,
+        }
+    }
+}
+
+impl TryFrom<&DynamicallyTypedData> for TradableKittyData {
+    type Error = TradeableKittyError;
+    fn try_from(a: &DynamicallyTypedData) -> Result<Self, Self::Error> {
+        a.extract::<TradableKittyData>()
+            .map_err(|_| TradeableKittyError::BadlyTyped)
+    }
+}
+
+impl UtxoData for TradableKittyData {
+    const TYPE_ID: [u8; 4] = *b"tdkt";
+}
+
+/// Reasons that tradable kitty opertaion may go wrong.
+#[derive(
+    Serialize,
+    Deserialize,
+    PartialEq,
+    Eq,
+    PartialOrd,
+    Ord,
+    Clone,
+    Encode,
+    Decode,
+    Hash,
+    Debug,
+    TypeInfo,
+)]
+pub enum TradeableKittyError {
+    /// Error in the underlying money piece.
+    MoneyError(money::ConstraintCheckerError),
+    /// Error in the underlying kitty piece.
+    KittyError(kitties::ConstraintCheckerError),
+    /// Dynamic typing issue.
+    /// This error doesn't discriminate between badly typed inputs and outputs.
+    BadlyTyped,
+    /// Input missing for the transaction.
+    InputMissingError,
+    /// Not enough amount to buy a kitty.
+    InsufficientCollateralToBuyKitty,
+    /// The number of input vs number of output doesn't match for a transaction.
+    NumberOfInputOutputMismatch,
+    /// Can't buy more than one kitty at a time.
+    CannotBuyMoreThanOneKittyAtTime,
+    /// Kitty basic properties such as DNA, free breeding, and a number of breedings, are altered error.
+    KittyBasicPropertiesAltered,
+    /// Kitty not available for sale.Occur when price is None.
+    KittyNotForSale,
+    /// Kitty price cant be none when it is available for sale.
+    KittyPriceCantBeNone,
```

Comment:
This is another clue that the field should not have been an `Option`. You noticed it yourself :)

Reply:
Yes made the price as mandatory.